hazelcast/src/main/java/com/hazelcast/instance/impl/Node.java

```diff
@@ -570,6 +570,7 @@ public void shutdown(final boolean terminate) {
             if (state != NodeState.SHUT_DOWN) {
                 shuttingDown.compareAndSet(true, false);
             }
+            loggingService.shutdown();
```

Comment:
Do we really want to call this in shutdown? Won't this shutdown the logging framework for the whole JVM? what if something wants to log afterwards?
Two cases I am thinking of:
- our tests where we shutdown nodes
- environment where the instance is managed by another framework, e.g. Spring

Reply:
I also wonder what shuting down log4j means. Will it shut down logger for whole jvm, including other libraries that uses it.

Reply:
> I suppose it shut downs the only Hazelcast's logger context.

That was my thought too. the `LogManager#shutdown` javadoc reads:

> Shutdown using the LoggerContext appropriate for the caller of this method.

Digging to figure out what is a "LoggerContext appropriate for the caller of this method", the only concrete information I could find was that a LoggerContext can be associated with a specific webapp's `ServletContext` (when running log4j under a webapp server), but no idea how `LoggerContext`s are created in a normal Java app.

I did a bit of research into [spring-boot](https://github.com/spring-projects/spring-boot/blob/c4de86c244acdcff69ed0aecacd254399be79ce2/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/logging/log4j2/SpringBootPropertySource.java#L37-L38) and [elasticsearch](https://github.com/elastic/elasticsearch/blob/main/distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/SystemJvmOptions.java#L54). They both disable the log4j shutdown hook and shut down log4j explicitly ([spring-boot](https://github.com/spring-projects/spring-boot/blob/7c77e1bb85b94d15b6e33477711a4f2e1ba14cf1/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/logging/log4j2/Log4J2LoggingSystem.java#L452-L454), [elasticsearch](https://github.com/elastic/elasticsearch/blob/c0bf49a9568eb333a927269cce31fd0b426358dc/server/src/main/java/org/elasticsearch/bootstrap/Elasticsearch.java#L482) ) , however it is rather safe to assume spring-boot/elasticsearch is probably the only app in the JVM.

I think we should control explicit shutdown of log4j (or any logging framework) by a `HazelcastProperty` and ensure it is enabled from our scripts (assuming Hazelcast is the only JVM app) but have it disabled by default. In this way, when using the client or embedded member, the logging framework explicit shutdown won't be executed by default since it can have unexpected side-effects. wdyt @frant-hartm @srknzl ?

Reply:
makes sense imo

Reply:
Implemented in https://github.com/hazelcast/hazelcast/pull/26006/commits/5c205295b78a0c39a6822b6c632a167cfce0b8de